### PR TITLE
fix: Deal with code smells

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Deal with code smells

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public abstract class AbstractCaller : ICaller
     protected IRequestMessage RequestMessage => _requestMessage ??= ServiceProvider.GetRequiredService<IRequestMessage>();
     protected IResponseMessage ResponseMessage => _responseMessage ??= ServiceProvider.GetRequiredService<IResponseMessage>();
 
-    public AbstractCaller(IServiceProvider serviceProvider)
+    protected AbstractCaller(IServiceProvider serviceProvider)
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs:16]: Processed

File: src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/Entities/Entity.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public bool Equals(Entity? other)
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/Entities/Entity.cs:24]: Processed

File: src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs [hunk 3]
```diff
@@ -310,14 +307,11 @@ protected virtual string GetUrl(string? url, IEnumerable<KeyValuePair<string, st
         return await SendAsync<TResponse>(request, cancellationToken);
     }
 
-    public virtual async Task<TResponse?> PatchAsync<TResponse>(
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs:313]: Processed

File: src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/Entities/Entity.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (doddgu) [reply src/BuildingBlocks/Ddd/Domain/Masa.BuildingBlocks.Ddd.Domain/Entities/Entity.cs:24]: No, you haven't fix it yet.

**Review by doddgu — CHANGES_REQUESTED** [review]

**Pull request merged**